### ISSUE TITLE
System hangs completly when auto complete suggestions are requested

### ISSUE THREAD

**Issue by NevilleAntony98** [benwaffle/vala-language-server#83]
**Describe the bug**
The system hangs completely while loading auto complete suggestion. The memory usage increased rapidly in the case of a simple hello world program.

**Software**
OS and version (e.g. Ubuntu 20.04): Arch linux
Code editor (e.g. VSCode): VSCode
Vala Language Server (e.g. git commit, or PPA/AUR version): 82a6e2b
Vala version (`valac --version`): 0.48.2

**To Reproduce**
Source code repo: A simple hello world program
```
public static int main (string[] args) {
        print ("Hello World!");
        return 0;
}
```

1. Open project
2. Try some statements to prompt auto complete suggestions

**Comment by Prince781** [comment]
I've determined this is a bug in upstream. Run `valac --keep-going hang.vala` on the following:

```
// hang.vala
public static int main (string[] args) {
        print ("Hello World!");
        p
        return 0;
}
```

gives you an endless stream of:
```
...
hang.vala:4.9-4.14: error: syntax error, following expression/statement delimiter `;' missing
        return 0;
        ^^^^^^
hang.vala:4.9-4.14: error: syntax error, following expression/statement delimiter `;' missing
        return 0;
        ^^^^^^
hang.vala:4.9-4.14: error: syntax error, following expression/statement delimiter `;' missing
        return 0;
        ^^^^^^
hang.vala:4.9-4.14: error: syntax error, following expression/statement delimiter `;' missing
        return 0;
...
```

**Comment by Prince781** [comment]
It turns out this case is very specific, and minor changes avoid the infinite loop. For example, if you have just `return;` instead of `return 0;`. However this still is a serious bug.

**Comment by Prince781** [comment]
I will fix this very very soon, but probably not until tomorrow or the day after as I'm busy with other stuff.

**Comment by Prince781** [comment]
Looks like a bug has been submitted upstream: https://gitlab.gnome.org/GNOME/vala/issues/961

**Comment by Prince781** [comment]
The bug has now been fixed in [Vala 0.48.3](https://gitlab.gnome.org/GNOME/vala/-/blob/0.48.3/NEWS#L17). You should no longer see this bug after updating your copy of Vala.
